spring-cloud-consul-discovery/src/test/java/org/springframework/cloud/consul/discovery/ConsulDiscoveryClientServerListQueryTagTests.java

```diff
@@ -0,0 +1,112 @@
+ * Copyright 2013-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.consul.discovery;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import com.ecwid.consul.v1.ConsulClient;
+import com.ecwid.consul.v1.agent.model.NewService;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cloud.client.ServiceInstance;
+import org.springframework.cloud.consul.test.ConsulTestcontainers;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.MOCK;
+import static org.springframework.cloud.consul.discovery.ConsulDiscoveryClientServerListQueryTagTests.NAME;
+
+/**
+ * Integration test to verify the
+ * {@link ConsulDiscoveryProperties#getServerListQueryTags()} is respected.
+ *
+ * @author Chris Bono
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = MOCK, classes = ConsulDiscoveryClientServerListQueryTagTests.TestConfig.class,
+		properties = { "spring.application.name=" + NAME,
+				"spring.cloud.consul.discovery.catalogServicesWatch.enabled=false",
+				"spring.cloud.consul.discovery.server-list-query-tags[" + NAME + "]=uat",
+				"spring.cloud.consul.discovery.defaultQueryTag=intg" })
+@DirtiesContext
+@ContextConfiguration(initializers = ConsulTestcontainers.class)
+public class ConsulDiscoveryClientServerListQueryTagTests {
+
+	public static final String NAME = "consulServiceServerListQueryTags";
+
+	@Autowired
+	private ConsulDiscoveryClient discoveryClient;
+
+	@Autowired
+	private ConsulClient consulClient;
+
+	private NewService intgService = serviceForEnvironment("intg", 9081);
+
+	private NewService uatService = serviceForEnvironment("uat", 9080);
+
+	@Before
+	public void setUp() {
+		this.consulClient.agentServiceRegister(this.intgService);
```

Comment:
Please do not use `this` where unnecessary. I know it's confusing, cause we have it elsewhere, but we're trying to modify it and at least not add it in new code.

Reply:
I am not a fan of `this` personally so I am excited about that. Is there a plan to update the formatter? Also, is this a "Spring-wide" initiative, or "Spring Cloud"? I just want to know because as I bounce around projects (Spring Boot in particular) it will be confusing to keep track of who does/not expect it.

Reply:
Unfortunately, it's just a Spring Cloud initiative. Our formatting differs slightly from team to team. In fact, you should not get formatting issues for that while building cloud projects. If you are, please let me know - we'll need to fix this.